src/SurveyDesign.jl

```diff
@@ -78,20 +78,25 @@ struct SurveyDesign <: AbstractSurveyDesign
         sampsize_labels = :sampsize
         data[!, sampsize_labels] = fill(length(unique(data[!, cluster])), (nrow(data),))
         if !(typeof(popsize) <: Nothing)
-            data[!, :weights] = data[!, popsize] ./ data[!, sampsize_labels]
-        elseif !(typeof(weights) <: Nothing)
-            data.weights = data[!, weights]
+            weights_labels = :weights
+            data[!, weights_labels] = data[!, popsize] ./ data[!, sampsize_labels]
+        elseif typeof(weights) <: Symbol
+            if !(typeof(data[!, weights]) <: Vector{<:Real})
+                error(string("given weights column ", weights , " is not of numeric type"))
```

Comment:
Following the discussion on `_weights`, I suggest removing this check and just keep the name `_weights` in all cases. However, I see the usefulness of having a specific error message in this case. I think it's better to use [`throw`](https://docs.julialang.org/en/v1/base/base/#Core.throw) with a specific exception - in this case either [`ArgumentError`](https://docs.julialang.org/en/v1/base/base/#Core.ArgumentError) or [`TypeError`](https://docs.julialang.org/en/v1/base/base/#Core.TypeError), although I think `TypeError` is appropriate in the case when the argument passed to the function has the wrong type, which is not the case here. `weights` was passed as a `Symbol`, it's just that it was referring to the wrong column.

Reply:
This would mean something like:
```julia
throw(ArgumentError(string("given weights column ", weights , " is not of numeric type")))
```

Reply:
Okay, i didnt know this is proper way of throwing errors, will use this syntax everywhere from now on.